www/include/configuration/configObject/host_template_model/formHostTemplateModel.php

```diff
@@ -43,12 +43,20 @@
 #
 $host = array();
 $macroArray = array();
-if (($o == "c" || $o == "w") && $host_id) {
+// Used to store all macro passwords
+$macroPasswords = array();
+
+define('PASSWORD_REPLACEMENT_VALUE', '**********');
+
+if (($o == HOST_TEMPLATE_MODIFY || $o == HOST_TEMPLATE_WATCH) && $host_id) {
     if (isset($lockedElements[$host_id])) {
-        $o = "w";
+        $o = HOST_TEMPLATE_WATCH;
     }
-    $DBRESULT = $pearDB->query("SELECT * FROM host, extended_host_information ehi WHERE host_id = '".$host_id."' AND ehi.host_host_id = host.host_id LIMIT 1");
-
+    $DBRESULT = $pearDB->query(
+        "SELECT * FROM host, extended_host_information ehi
+        WHERE host_id = " . ((int) $host_id)
+        . " AND ehi.host_host_id = host.host_id LIMIT 1"
+    );
```

Comment:
Shouldn't you use a prepare execute statement with `?` for this and following queries ?

Reply:
There is no PDO in PHP 5.4

Reply:
As @jeremyjaouen suggestion, it is possible to use `?` to bind values, which works with PearDB